packer/linux/conf/bin/bk-fetch.sh

```diff
@@ -1,13 +1,66 @@
 #!/bin/bash
 set -euo pipefail
 
+# fetch_ssm_parameters fetches all SSM parameters under the given path and writes them to the given output file
+fetch_ssm_parameters() {
+  local ssm_path="$1"
+  local output_file="$2"
+
+  # check if the ssm_path is set
+  if [ -z "$ssm_path" ]; then
+    echo "ssm_path is not set"
+    return 1
+  fi
+
+  # check if the output_file is set
+  if [ -z "$output_file" ]; then
+    echo "output_file is not set"
+    return 1
+  fi
+
+  # trim off ssm: prefix
+  ssm_path=${ssm_path//ssm:/}
+
+  # Get all SSM parameter names under the given path
+  #
+  # NOTE: The maximum number of parameters that can be retrieved is 25 to avoid throttling
+  # in the case of misconfigured SSM path with a large number of child parameters
+  local ssm_parameter_names
+  ssm_parameter_names=$(aws ssm get-parameters-by-path \
+    --path "$ssm_path" \
+    --recursive \
+    --max-items 25 \
+    --with-decryption \
+    --query 'Parameters[].Name' \
+    --output text)
+
+  # Loop through each parameter and export it as an environment variable
+  for name in $ssm_parameter_names; do
+    local value
+    value=$(aws ssm get-parameter \
+      --name "$name" \
+      --with-decryption \
+      --query 'Parameter.Value' \
+      --output text)
+    if [ -n "$name" ] && [ -n "$value" ]; then
+      local var_name
+      var_name=$(echo "$name" | awk -F/ '{print toupper($NF)}')
+      echo "Exported variable: $var_name"
+      echo "$var_name=$value" >>"$output_file"
+    fi
+  done
```

Comment:
you can use a single `aws ssm` call to get all the information and take advantage of the fact that `jq` is installed as well to make this a lot more automated (at the cost of the `Exported variable` log) with something like:

```suggestion
  aws ssm get-parameters-by-path \
    --path "$ssm_path" \
    --recursive \
    --with-decryption \
    --query 'Parameters[*].[Name,Value]' \
    --output json \
  | jq -r '.Parameters[] | [(.Name | split("/")[-1] | ascii_upcase), (["\"", .Value, "\""] | join(""))] | join("=")' \
  > $output_file
```

This has the added benefit of not passing the variable name or value through bash variable interpolation which may be a security risk